core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -822,18 +827,43 @@ private[spark] object MapOutputTracker extends Logging {
     } {
       objOut.close()
     }
-    val arr = out.toByteArray
+
+    val arr: Array[Byte] = {
+      val compressedOut = new ByteArrayOutputStream(4096)
+      val zos = new ZstdOutputStream(compressedOut)
```

Comment:
Is the default compress level good for us?

Reply:
Better than the gzip. We can tune it later.